### PR TITLE
doc: update versions for NCS 1.7 release

### PR DESCRIPTION
**Pull request by greg-fer**
Updated files except VERSION for documentation NCS 1.7.0 release.
VERSION to be handled by Vestavind.
NCSDK-11022.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

-----

- [x] This PR to nrfxlib needs to go in before this one: https://github.com/nrfconnect/sdk-nrfxlib/pull/549
- [x] These PRs to sdk-nrf need to go in before this one: https://github.com/nrfconnect/sdk-nrf/pull/5534 and https://github.com/nrfconnect/sdk-nrf/pull/5475
- [x] Check TFM version with @SebastianBoe  -- updated to 1.3.99
- [x] Check MCUboot version with @carlescufi

### CODE REVIEW

File: doc/versions.json [hunk 1]
```diff
@@ -20,8 +21,17 @@
...
     "latest": {
-      "ncs": "1.6.99",
-      "nrfxlib": "1.6.99",
+      "ncs": "1.7.0",
+      "nrfxlib": "1.7.0",
+      "zephyr": "2.6.99",
+      "mcuboot": "1.7.99",
+      "nrfx": "2.5",
+      "tfm": "1.3.99",
+      "matter": ""
```
Comment (greg-fer) [inline doc/versions.json:30]: @carlescufi , @tejlmand , kindly please confirm versions for 1.7.0...

Comment (carlescufi) [reply doc/versions.json:30]: Yes, except matter which needs comment from @LuDuda

Comment (LuDuda) [reply doc/versions.json:30]: As discussed internally, let's not specify any versioning for now.

File: doc/nrf/index.rst [hunk 2]
```diff
@@ -10,7 +10,8 @@ The |NCS| includes the Zephyr™ real-time operating system (RTOS), which is bui
 
 .. note::
 
-   This version of the |NCS| supports prototyping and evaluation, but should not be used for production and deployment in end products.
+   nRF53 Series devices are supported for production.
+   Wireless protocol stacks and libraries may indicate support for development or support for production for different series of devices based on verification and certification status.
```
Comment (greg-fer) [inline doc/nrf/index.rst:14]: @umapraseeda , updated as suggested, but here :)

File: doc/versions.json [hunk 3]
```diff
@@ -20,12 +21,21 @@
...
     "latest": {
-      "ncs": "1.6.99",
-      "nrfxlib": "1.6.99",
+      "ncs": "1.7.0",
+      "nrfxlib": "1.7.0",
       "zephyr": "2.6.99",
       "mcuboot": "1.7.99",
       "nrfx": "2.5",
-      "tfm": "1.3.99",
+      "tfm": "1.3.0",
```
Comment (umapraseeda) [inline doc/versions.json:29]: @gmarull, to make the tfm versions identical, in addition to this change, should we make changes to tfm/conf.py also (like set _version = "1.7.0"_)

Comment (gmarull) [reply doc/versions.json:29]: do you guys want to make the changes in this PR?

Comment (greg-fer) [reply doc/versions.json:29]: I would prefer a separate PR.

Comment (gmarull) [reply doc/versions.json:29]: https://github.com/nrfconnect/sdk-nrf/pull/5616

**Comment by umapraseeda** [comment]
@greg-fer, As discussed offline, please add the same note in https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/applications/asset_tracker_v2/doc/asset_tracker_v2_description.html#setup after this section also: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_nrf91_features.html#predicted-gps-p-gps. @martiles the discussed change will be added in this PR.

**Review by umapraseeda — APPROVED** [review]

**Review by simensrostad — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Review by shantha-14 — APPROVED** [review]

File: doc/nrf/links.txt [hunk 4]
```diff
@@ -342,11 +342,11 @@
 
 .. ### Source: www.segger.com
 
-.. _`SEGGER Embedded Studio (Nordic Edition) - Windows x86`: https://www.segger.com/downloads/embedded-studio/embeddedstudio_arm_nordic_win_x86
-.. _`SEGGER Embedded Studio (Nordic Edition) - Windows x64`: https://www.segger.com/downloads/embedded-studio/embeddedstudio_arm_nordic_win_x64
-.. _`SEGGER Embedded Studio (Nordic Edition) - Mac OS x64`: https://www.segger.com/downloads/embedded-studio/embeddedstudio_arm_nordic_macos
-.. _`SEGGER Embedded Studio (Nordic Edition) - Linux x86`: https://www.segger.com/downloads/embedded-studio/embeddedstudio_arm_nordic_linux_x86
-.. _`SEGGER Embedded Studio (Nordic Edition) - Linux x64`: https://www.segger.com/downloads/embedded-studio/embeddedstudio_arm_nordic_linux_x64
```
Comment (gmarull) [inline doc/nrf/links.txt:349]: these changes should likely go into a separate commit (it's not about versions update as said in b697289b4d110b46747945d07e7107dc183e548b)

File: doc/versions.json [hunk 5]
```diff
@@ -35,7 +35,7 @@
...
```
Comment (gmarull) [inline doc/versions.json]: tfm is later changed to `1.3.99`, why is changed to `1.3.0` here? commit mentions nrfxlib: 00347b8b60f7527311ffcfcdd2994a78842d8cca

File: doc/nrf/links.txt (hunk 4 above)
Comment (greg-fer) [reply doc/nrf/links.txt:349]: This is how we always did it. See https://github.com/nrfconnect/sdk-nrf/pull/4938/. Also, it is required in the related JIRA. We can do it differently starting with the next major release.

File: doc/versions.json (hunk 5 above)
Comment (greg-fer) [reply doc/versions.json]: Fixed.

**Review by gmarull — APPROVED** [review]
